src/XMakeBuildEngine/Definition/ProjectCollection.cs

```diff
@@ -384,7 +384,7 @@ public static Version Version
                     // Use .CodeBase instead of .Location, because .Location doesn't
                     // work when Microsoft.Build.dll has been shadow-copied, for example
                     // in scenarios where NUnit is loading Microsoft.Build.
-                    s_engineVersion = new Version(FileVersionInfo.GetVersionInfo(FileUtilities.ExecutingAssemblyPath).ProductVersion);
+                    s_engineVersion = new Version(FileVersionInfo.GetVersionInfo(FileUtilities.ExecutingAssemblyPath).FileVersion);
```

Comment:
It was something like `15.1.0+g1a4c5a01a8` which crashed Version ctor. I would hope people aren't using it in that way, but can't really be sure :disappointed: